packages/components/src/form-token-field/index.tsx

```diff
@@ -406,15 +410,15 @@ export function FormTokenField( props: FormTokenFieldProps ) {
 		}
 	}
 
-	function addCurrentToken() {
+	function addCurrentToken( isOnBlur = false ) {
```

Comment:
the `isOnBlur` param is needed as the `setIsActive( false );` in the `onBlur` function is registered until the next render so `isActive` is still `true` when `addNewToken` is called. There may be a better way of achieving this but couldn't think of one.

Reply:
We could use `useEffect` and react to `isActive` becoming `false`, but the way this component is written is quite entangled, and therefore we'd still face issues.

I wonder if we could use the `hasFocus()` function and the `__experimentalAddOnBlur` prop in the `addNewToken` function instead of passing a `isOnBlur` argument around?

Reply:
[Refactored this here](https://github.com/WordPress/gutenberg/pull/53976/commits/bf20870aec7677b9f194baf4f1fecd14f2c6ac1a) to get rid of `isOnBlur` as suggested and it seems to work, but will do some more testing tomorrow to make sure I didn't miss any edge cases.